### DISCUSSION TITLE
"Undo" underline

### DISCUSSION THREAD

**Discussion by yochem** [typst/typst#2883] [Q&A]
Related to #420.

I have `#show link: underline.with(stroke: gray)` on top of my document, but I don't want the underline in the outline (but do want to keep the links of course).

How would I reset this underlining? I tried to use a `#show link: underline.with(<transparent line>)` hack, but couldn't get a transparent underline to work.

Any other ideas?

**Comment by WalrusGumboot** [comment]
Could you share a MWE? I can't seem to reproduce the formatting you get with the following code:
```typst
#set page(width: 30em)

#show link: underline.with(stroke: gray)

#outline()

= Heading
#lorem(30)
== Subheading
#lorem(30)

#link("https://www.typst.app/", [A link in text])
```
![image](https://github.com/typst/typst/assets/39743643/07365d97-240f-483a-a1a6-9b337b02fbb3)

**Comment by yochem** [comment]
Oops, forgot an important detail. I redefined the look of outline entries, and used `link(it.element.location(), ...)` to still make them clickable. Should I use another function for this? Here's a MWE of the current setup:


```typst
#set page(width: 30em)


#show link: underline.with(stroke: gray)

#show outline.entry: it => {
  link(it.element.location(), [\- #it.body])
}

#outline()

= Heading
#lorem(30)
== Subheading
#lorem(30)

#link("https://www.typst.app/", [A link in text])
```
![test](https://github.com/typst/typst/assets/23235841/259a481a-c18e-4476-9f87-232cd035244c)

**Reply by laurmaedje** [reply]
You could write a link show rule that only underlines specific links:
```typst
#show link: it => {
  if type(it.dest) == str {
    underline(it)
  } else {
    it
  }
}
```
This would only underline links to URLs, but you can adjust the condition to your liking.

For the future, it is also planned to make show rules revokable, but that will still take some time.

**Reply by yochem** [reply]
Thank you for your response. But I do like that `@heading` gives an underlined link to the heading section, just not in the toc. I was hoping there was a simple solution to it, but I'll probably use a show rule for links that checks if the current context is TOC.

**Reply by yochem** [reply]
Btw, I also noticed that you need show rules for both `ref` and `cite` if you want to underline both textual citations (`cite`) and normal citations (`ref`). Which feels unexpected.
